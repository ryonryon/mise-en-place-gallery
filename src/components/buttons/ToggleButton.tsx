import styled from "@emotion/styled";
import * as React from "react";
import { secondary, white } from "../../constants/colors";

const ToggleButtonDiv = styled.div`
  background-color: ${white};
  display: inline-block;
  border-radius: 5px;
  border: solid 0.5px ${secondary};
  color: ${secondary};
  cursor: pointer;
  margin: 16px;

  div {
    display: inline-block;
    padding: 10px 12px;
    border-right: solid 0.3px ${secondary};

    &:last-child {
      border-style: none;
    }

    &:hover {
      background-color: rgba(221, 221, 221, 0.5);
    }
  }
`;

interface Props extends React.HTMLAttributes<HTMLDivElement> {
  values: string[];
}

const ToggleButton = ({ values, ...rest }: Props) => (
  <ToggleButtonDiv>
    {values.map((item, index) => (
      <div key={index} {...rest}>
        {item}
      </div>
    ))}
  </ToggleButtonDiv>
);

ToggleButton.defaultProps = {
  values: ["item 1", "item 2", "item 3", "item 4"]
};

export default ToggleButton;
