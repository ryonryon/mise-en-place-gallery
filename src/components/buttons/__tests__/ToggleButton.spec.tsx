import * as React from "react";
import * as renderer from "react-test-renderer";

import ToggleButton from "../ToggleButton";

describe("ToggleButton test", () => {
  it("tests ToggleButton renders correctly ", () => {
    const toggleButton = renderer.create(<ToggleButton />).toJSON();
    expect(toggleButton).toMatchSnapshot();
  });
});
