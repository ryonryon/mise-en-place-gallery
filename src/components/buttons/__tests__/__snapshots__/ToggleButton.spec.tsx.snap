// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton test tests ToggleButton renders correctly  1`] = `
<div
  className="css-1cm4m5i"
>
  <div>
    item 1
  </div>
  <div>
    item 2
  </div>
  <div>
    item 3
  </div>
  <div>
    item 4
  </div>
</div>
`;
